internal/pkg/otel/configtranslate/otelconfig.go

```diff
@@ -0,0 +1,366 @@
+
+	otelcomponent "go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/pipeline"
+	"golang.org/x/exp/maps"
+
+	elasticsearchtranslate "github.com/elastic/beats/v7/libbeat/otelbeat/oteltranslate/outputs/elasticsearch"
+	"github.com/elastic/beats/v7/x-pack/filebeat/fbreceiver"
+	"github.com/elastic/beats/v7/x-pack/libbeat/management"
+	"github.com/elastic/beats/v7/x-pack/metricbeat/mbreceiver"
+	"github.com/elastic/elastic-agent-client/v7/pkg/client"
+	"github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/application/info"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/application/paths"
+	"github.com/elastic/elastic-agent/pkg/component"
+	"github.com/elastic/elastic-agent/pkg/component/runtime"
+)
+
+// This is a prefix we add to all names of Otel entities in the configuration. Its purpose is to avoid collisions with
+// user-provided configuration
+const OtelNamePrefix = "_agent-component/"
+
+type exporterConfigTranslationFunc func(*config.C) (map[string]any, error)
+
+var (
+	OtelSupportedOutputTypes         = []string{"elasticsearch"}
+	OtelSupportedInputTypes          = []string{"filestream"}
+	configTranslationFuncForExporter = map[otelcomponent.Type]exporterConfigTranslationFunc{
+		otelcomponent.MustNewType("elasticsearch"): translateEsOutputToExporter,
+	}
+)
+
+// GetOtelConfig returns the Otel collector configuration for the given component model.
+// All added component and pipelines names are prefixed with OtelNamePrefix.
+// Unsupported components are quietly ignored.
+func GetOtelConfig(model *component.Model, info info.Agent) (*confmap.Conf, error) {
+	components := getSupportedComponents(model)
+	if len(components) == 0 {
+		return nil, nil
+	}
+	otelConfig := confmap.New() // base config, nothing here for now
+
+	for _, comp := range components {
+		componentConfig, compErr := getCollectorConfigForComponent(comp, info)
+		if compErr != nil {
+			return nil, compErr
+		}
+		// the assumption here is that each component will define its own receivers, and the shared exporters
+		// will be merged
+		mergeErr := otelConfig.Merge(componentConfig)
+		if mergeErr != nil {
+			return nil, fmt.Errorf("error merging otel config for component %s: %w", comp.ID, mergeErr)
+		}
+	}
+	return otelConfig, nil
+}
+
+// IsComponentOtelSupported checks if the given component can be run in an Otel Collector.
+func IsComponentOtelSupported(comp *component.Component) bool {
+	return slices.Contains(OtelSupportedOutputTypes, comp.OutputType) &&
+		slices.Contains(OtelSupportedInputTypes, comp.InputType)
+}
+
+// getSupportedComponents returns components from the given model that can be run in an Otel Collector.
+func getSupportedComponents(model *component.Model) []*component.Component {
+	var supportedComponents []*component.Component
+
+	for _, comp := range model.Components {
+		if IsComponentOtelSupported(&comp) {
+			comp := comp
```

Comment:
I don't think this loop var assignment is needed after Go 1.22 https://go.dev/wiki/LoopvarExperiment

Reply:
It's not, but we have a linter enabled (exportloopref) that flags it. I'd rather disable that linter and fix all the incidences of this in a separate PR.